diff --git a/date_all_day/date_all_day.module b/date_all_day/date_all_day.module
index 93fc61b..eb4f5aa 100644
--- a/date_all_day/date_all_day.module
+++ b/date_all_day/date_all_day.module
@@ -407,13 +407,5 @@ function date_all_day_date_field_widget_settings_form_alter(&$form, $context) {
     '#title' => t('Display all day checkbox'),
     '#default_value' => $settings['display_all_day'],
     '#description' => t("Determines whether to display the 'All Day' checkbox to the user."),
-    '#weight' => 8,
-    '#fieldset' => 'date_format',
   );
-
-  // Hide the option to use the all day checkbox for dates with no time.
-  if (!date_has_time($field['settings']['granularity'])) {
-    $form['display_all_day']['#type'] = 'hidden';
-    $form['display_all_day']['#value'] = 0;
-  }
 }
