diff --git a/double_field.module b/double_field.module
index 25fee6c..02e133c 100644
--- a/double_field.module
+++ b/double_field.module
@@ -213,9 +213,14 @@ function double_field_field_validate($entity_type, $entity, $field, $instance, $
   list($widget['first'], $widget['second']) = explode('_&_', $instance['widget']['type']);
   foreach ($items as $delta => $item) {
 
-    // Checkboxes get value from widget settigns.
+    // Checkboxes get value from widget settings.
     foreach (array('first', 'second') as $subfield) {
-      $item[$subfield] = ($widget[$subfield] == 'checkbox' && $item[$subfield] == $settings[$subfield]['checkbox']['off_value']) ? '' : $item[$subfield];
+      if (isset($item[$subfield])) {
+        $item[$subfield] = ($widget[$subfield] == 'checkbox' && $item[$subfield] == $settings[$subfield]['checkbox']['off_value']) ? '' : $item[$subfield];
+      }
+      else {
+        $item[$subfield] = '';
+      }
     }
 
     if ($item['first'] !== '' || $item['second'] !== '') {
