diff --git a/youtube.module b/youtube.module
index 876b3df..eafddeb 100644
--- a/youtube.module
+++ b/youtube.module
@@ -397,7 +397,7 @@ function youtube_field_formatter_view($entity_type, $entity, $field, $instance,
         $element[$delta] = array(
           '#theme' => 'youtube_video',
           '#video_id' => $item['video_id'],
-          '#entity_title' => $entity->title,
+          '#entity_title' => !empty($entity->title) ? check_plain($entity->title) : NULL,
           '#size' => array_key_exists('youtube_size', $display['settings']) ? $display['settings']['youtube_size']: NULL,
           '#width' => array_key_exists('youtube_width', $display['settings']) ? $display['settings']['youtube_width'] : NULL,
           '#height' => array_key_exists('youtube_height', $display['settings']) ? $display['settings']['youtube_height'] : NULL,
@@ -451,7 +451,7 @@ function youtube_field_formatter_view($entity_type, $entity, $field, $instance,
         $element[$delta] = array(
           '#theme' => 'youtube_thumbnail',
           '#video_id' => $item['video_id'],
-          '#entity_title' => $entity->title,
+          '#entity_title' => !empty($entity->title) ? check_plain($entity->title) : NULL,
           '#image_style' => $display['settings']['image_style'],
           '#image_link' => isset($uri) ? $uri : '',
         );
diff --git a/youtube.theme.inc b/youtube.theme.inc
index 3715f80..5df644f 100644
--- a/youtube.theme.inc
+++ b/youtube.theme.inc
@@ -76,9 +76,12 @@ function theme_youtube_video($variables) {
   $path = 'https://www.' . $domain . '/embed/' . $id;
   $src = url($path, array('query' => $query));
 
-  $player_title = t('Embedded video for @entity_title', array(
-    '@entity_title' => $variables['entity_title'],
-  ));
+  $player_title = t('Embedded video');
+  if (!empty($variables['entity_title'])) {
+    $player_title .= t(' for @entity_title', array(
+      '@entity_title' => $variables['entity_title'],
+    ));
+  }
 
   $output = '<iframe id="' . drupal_html_id($player_class) . '" class="' . $player_class . '"
     width="' . $dimensions['width'] . '" height="' . $dimensions['height'] . '"
@@ -118,9 +121,12 @@ function theme_youtube_thumbnail($variables) {
     }
   }
 
-  $alt = t('Embedded thumbnail for @entity_title', array(
-    '@entity_title' => $variables['entity_title'],
-  ));
+  $alt = t('Embedded thumbnail');
+  if (!empty($variables['entity_title'])) {
+    $alt .= t(' for @entity_title', array(
+      '@entity_title' => $variables['entity_title'],
+    ));
+  }
 
   if ($style) {
     $uri = 'public://' . $youtube . '/' . $id . '.png';
